gnucash/report/trep-engine.scm

```diff
@@ -1226,6 +1226,16 @@ be excluded from periodic reporting.")
                                   "number-cell"
                                   (xaccSplitGetAmount split)))))
 
+               (add-if 'association-column?
+                       (vector ""
+                               (lambda (split transaction-row?)
+                                 (let ((url (xaccTransGetAssociation (xaccSplitGetParent split))))
+                                   (if (not (string-null? url))
+                                     (gnc:make-html-table-cell/markup
+                                      "text-cell"
+                                      (gnc:html-transaction-association-anchor (xaccSplitGetParent split) (_ "Association")))
+                                       " ")))))
+
```

Comment:
That's rather impressive. Here's a follow-up commit that you can fixup to this one. Minor nitpick: Wouldn't `trans-association-guid` be more consistent?

```
modified   gnucash/report/trep-engine.scm
@@ -937,6 +937,7 @@ be excluded from periodic reporting.")
       (list (N_ "Other Account Code")           "j"  (_ "Display the other account code?") #f)
       (list (N_ "Shares")                       "k"  (_ "Display the number of shares?") #f)
       (list (N_ "Price")                        "l"  (_ "Display the shares price?") #f)
+      (list (N_ "Association")                  "l5" (_ "Display the transaction association") #f)
       ;; note the "Amount" multichoice option in between here
       (list optname-grid                        "m5" (_ "Display a subtotal summary table. This requires Display/Amount being 'single") #f)
       (list (N_ "Running Balance")              "n"  (_ "Display a running balance?") #f)
@@ -1061,6 +1062,7 @@ be excluded from periodic reporting.")
                      (opt-val gnc:pagename-display (N_ "Other Account Name"))))
           (cons 'shares (opt-val gnc:pagename-display (N_ "Shares")))
           (cons 'price (opt-val gnc:pagename-display (N_ "Price")))
+          (cons 'association (opt-val gnc:pagename-display "Association"))
           (cons 'amount-single (eq? amount-setting 'single))
           (cons 'amount-double (eq? amount-setting 'double))
           (cons 'common-currency (opt-val gnc:pagename-general optname-common-currency))
@@ -1226,15 +1228,17 @@ be excluded from periodic reporting.")
                                   "number-cell"
                                   (xaccSplitGetAmount split)))))
 
-               (add-if 'association-column?
+               (add-if (column-uses? 'association)
                        (vector ""
                                (lambda (split transaction-row?)
-                                 (let ((url (xaccTransGetAssociation (xaccSplitGetParent split))))
-                                   (if (not (string-null? url))
-                                     (gnc:make-html-table-cell/markup
-                                      "text-cell"
-                                      (gnc:html-transaction-association-anchor (xaccSplitGetParent split) (_ "Association")))
-                                       " ")))))
+                                 (let ((url (xaccTransGetAssociation
+                                             (xaccSplitGetParent split))))
+                                   (and (not (string-null? url))
+                                        (gnc:make-html-table-cell/markup
+                                         "text-cell"
+                                         (gnc:html-transaction-association-anchor
+                                          (xaccSplitGetParent split)
+                                          (_ "📎"))))))))
 
                (add-if (column-uses? 'price)
                        (vector (_ "Price")
```

Reply:
@christopherlam Thanks for the update and the suggested change which I have done.